core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -429,6 +435,7 @@ private void handleRequest(ChannelHandlerContext ctx, DecodedHttpRequest req) th
                     logger.warn("Unexpected exception:", t);
                 }
                 return null;
+                // Reschedule to abort the request after all post-processes such logging have been handled.
             }, eventLoop);
```

Comment:
I'm not sure about the reason why we didn't use `inEventLoop()` instead of using async method from the first, but how about using `inEventLoop()` check because mostly the future is completed by the `eventLoop`?

Reply:
At first, I tried to use `handle` but I found that some tests are failed because `AbortedStreamException` was logged in `RequestLog`.
This is because this callback is completed before the `RequestLog` ends with an specific cause.

Reply:
> some tests are failed because AbortedStreamException was logged in RequestLog.

Do they fail even after we introduce the `ResponseCompleteException`? 🤔

Reply:
Sorry, my analysis was incorrect.
The following test which intends to set `requestCause` to the `RuntimeException` is failed if we use `handle`. Because `requestCause` was set to `null`
https://github.com/line/armeria/blob/ed021f3d0fcb8160032a89f6a625bd2fbb0ae542/core/src/test/java/com/linecorp/armeria/server/logging/ContentPreviewingFailureTest.java#L50-L53

1. `tryComplete(RuntimeException)` is called.
https://github.com/line/armeria/blob/3c7d210f4cf0a1f4f061a1b24fd7d6e21e7c156d/core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java#L360-L363
2. The callback makes `req.abort(RuntimeException)` called.
https://github.com/line/armeria/blob/afe2c2e0979ab958ce755d3e58cd40ffe5d15a84/core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java#L419-L424
3. However, the `RuntimeException` is not propagated to 4 due to the following condition.
https://github.com/line/armeria/blob/4dd0304a7cfe3d2a9c88d4afb0f5cb9f4059fbcb/core/src/main/java/com/linecorp/armeria/common/stream/DefaultStreamMessage.java#L227-L232
4. The request-side log is completed successfully.
https://github.com/line/armeria/blob/afe2c2e0979ab958ce755d3e58cd40ffe5d15a84/core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java#L402-L405
5. endLogRequest(RuntimeException) is ignored since it has been completed by 4.
https://github.com/line/armeria/blob/3c7d210f4cf0a1f4f061a1b24fd7d6e21e7c156d/core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java#L360-L361